influxql/ast.go

```diff
@@ -177,6 +193,220 @@ func (a SortFields) String() string {
+	return buf.String()
+}
+
+// DropDatabaseStatement represents a command to drop a database.
+type DropDatabaseStatement struct {
+	// Name of the database to be dropped.
+	Name string
+}
+
+// String returns a string representation of the drop database statement.
+func (s *DropDatabaseStatement) String() string {
+	var buf bytes.Buffer
+	_, _ = buf.WriteString("DROP DATABASE ")
+	_, _ = buf.WriteString(s.Name)
+	return buf.String()
+}
+
+// CreateUserStatement represents a command for creating a new user.
+type CreateUserStatement struct {
+	// Name of the user to be created.
+	Name string
+
+	// User's password
+	Password string
+}
+
+// String returns a string representation of the create user statement.
+func (s *CreateUserStatement) String() string {
+	var buf bytes.Buffer
+	_, _ = buf.WriteString("CREATE USER ")
+	_, _ = buf.WriteString(s.Name)
+	_, _ = buf.WriteString(" WITH PASSWORD ")
+	_, _ = buf.WriteString(s.Password)
+	return buf.String()
+}
+
+// DropUserStatement represents a command for dropping a user.
+type DropUserStatement struct {
+	// Name of the user to drop.
+	Name string
+}
+
+// String returns a string representation of the drop user statement.
+func (s *DropUserStatement) String() string {
+	var buf bytes.Buffer
+	_, _ = buf.WriteString("DROP USER ")
+	_, _ = buf.WriteString(s.Name)
+	return buf.String()
+}
+
+// Privilege is a type of action a user can be granted the right to use.
+type Privilege int
+
+const (
+	ReadPrivilege Privilege = iota
+	WritePrivilege
+	AllPrivileges
+)
+
+// String returns a string representation of a Privilege.
+func (p Privilege) String() string {
+	switch p {
+	case ReadPrivilege:
+		return "READ"
+	case WritePrivilege:
+		return "WRITE"
+	case AllPrivileges:
+		return "ALL PRIVILEGES"
+	}
+	return ""
+}
+
+// GrantStatement represents a command for granting a privilege.
+type GrantStatement struct {
+	// The privilege to be granted.
+	Privilege Privilege
+
+	// Thing to grant privilege on (e.g., a DB).
+	On string
+
+	// Who to grant the privilege to.
+	User string
+}
+
+// String returns a string representation of the grant statement.
+func (s *GrantStatement) String() string {
+	var buf bytes.Buffer
+	_, _ = buf.WriteString("GRANT ")
+	_, _ = buf.WriteString(s.Privilege.String())
+	if s.On != "" {
+		_, _ = buf.WriteString(" ON ")
+		_, _ = buf.WriteString(s.On)
+	}
+	_, _ = buf.WriteString(" TO ")
+	_, _ = buf.WriteString(s.User)
+	return buf.String()
+}
+
+// RevokeStatement represents a command to revoke a privilege from a user.
+type RevokeStatement struct {
+	// Privilege to be revoked.
+	Privilege Privilege
+
+	// Thing to revoke privilege to (e.g., a DB)
+	On string
+
+	// Who to revoke privilege from.
+	User string
+}
+
+// String returns a string representation of the revoke statement.
+func (s *RevokeStatement) String() string {
+	var buf bytes.Buffer
+	_, _ = buf.WriteString("REVOKE ")
+	_, _ = buf.WriteString(s.Privilege.String())
+	if s.On != "" {
```

Comment:
How could ON be ""? Surely privileges need to be set on something? Or is there is a default?